### DISCUSSION TITLE
[@astrojs/image] Import SVG with same data as other images

### DISCUSSION THREAD

**Discussion by jkjustjoshing** [withastro/roadmap#280] [Proposal]
With `@astrojs/image` installed, if I have a `.png` file I want to use, I can do the following:
```javascript
import img from './path/to/image.png';

// img is now
// {
//    src: '/absolute/path/to/image.png',
//    width: 500,
//    height: 500,
//    format: 'svg'
// }
```
If I want to import an `.svg` file and use the path in an `<img>` element, I can do the following:
```javascript
import img from './path/to/image.svg';

// img is now
// '/some/path/to/image.svg'
```

There's no way right now to automatically add the SVG image's width and height to the `<img>` element, as is [recommended](https://css-tricks.com/do-this-to-improve-image-loading-on-your-website/).

It would be great if a `.png` and `.svg` imported in the same compatible format, though that would clearly be a breaking change for anyone importing SVGs. Alternatively, SVG imports could have width and height named exports.

**Comment by delucis** [comment]
Agreed — this already came up in discussions with @tony-sull (perhaps even adding optimisation with `svgo`). I'm sure we'll get to something like this before the image integration reaches v1.

**Comment by jkjustjoshing** [comment]
Another thing I'd love to see the integration do with SVGs:

If I add an `.svg` file to the `<Picture>` element with `formats={['avif', 'webp']}`, I'd like the image integration to convert the SVG file to avif/webp and use those versions in `<source>`. Treat SVG images the exact same as JPG images.

**Comment by jasikpark** [comment]
isn't a problem here that svgs don't have an intrinsic size?

**Reply by jkjustjoshing** [reply]
The `<Picture>` element requires a `widths` prop, and most SVGs passed to the <img> element are going to have a viewBox or width/height attributes. The optimization could error if those aren't found.
